cns/cniconflist/generator_linux.go

```diff
@@ -0,0 +1,75 @@
+package cniconflist
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/Azure/azure-container-networking/cni"
+	cninet "github.com/Azure/azure-container-networking/cni/network"
+	"github.com/Azure/azure-container-networking/cni/util"
+	"github.com/Azure/azure-container-networking/cns/configuration"
+	"github.com/Azure/azure-container-networking/network"
+)
+
+const (
+	portmapType            = "portmap"
+	portMappingsCapability = "portMappings"
+)
+
+var errUnsupportedScenario = errors.New("unsupported scenario")
+
+func NewGenerator(w io.WriteCloser, scenario configuration.CNIConflistScenario) (*Generator, error) {
+	g := Generator{writer: w}
+	switch scenario {
+	case configuration.CNIConflistScenarioV4Overlay:
+		g.scenario = scenario
+	default:
+		return nil, fmt.Errorf("can't build generator: %w", errUnsupportedScenario)
+	}
+
+	return &g, nil
+}
+
+func (g *Generator) Generate() error {
+	var azureVnetConfig cni.NetworkConfig
+	if g.scenario == configuration.CNIConflistScenarioV4Overlay {
+		azureVnetConfig = cni.NetworkConfig{
+			Type:              cniType,
+			Mode:              cninet.OpModeTransparent,
+			ExecutionMode:     string(util.V4Swift),
+			IPsToRouteViaHost: []string{nodeLocalDNSIP},
+			Ipam: cni.Ipam{
+				Type: network.AzureCNS,
+				Mode: string(util.V4Overlay),
+			},
+		}
+	}
+
+	portmap := struct {
+		Type         string          `json:"type"`
+		Capabilities map[string]bool `json:"capabilities"`
+		SNAT         bool            `json:"snat"`
+	}{
+		Type: portmapType,
+		Capabilities: map[string]bool{
+			portMappingsCapability: true,
+		},
+		SNAT: true,
+	}
```

Comment:
Is it worth reifying portmap into its own type where it's pretty common in CNI configs?

Reply:
the problem I ran into was that their conflist includes this `SNAT` bool that is not in the containernetworking/cni types, and IIRC they didn't export their config type. But I'll check on that part. If I can import their config type, that would be good.